promgen/static/js/promgen.vue.js

```diff
@@ -323,3 +326,133 @@ app.component('exporter-test', {
         }
     }
 });
+
+const silenceListStore = Vue.reactive({
+    state: {
+        show: false,
+        labels: []
+    },
+    addFilterLabel(label, value) {
+        const existingLabel = this.state.labels.find(item => item.label === label && item.value === value);
+        if (!existingLabel) {
+            this.state.labels.push({ label, value });
+        }
+    },
+    removeFilterLabel(label, value) {
+        const index = this.state.labels.findIndex(item => item.label === label && item.value === value);
+        if (index > -1) {
+            this.state.labels.splice(index, 1);
+        }
+    },
+    showModal() {
+        this.state.show = true;
+    },
+    hideModal() {
+        this.state.show = false;
+    },
+});
+
+app.component('silence-list-modal', {
+    template: '#silence-list-modal-template',
+    delimiters: ['[[', ']]'],
+    mixins: [mixins],
+    data() {
+        return {
+            state: silenceListStore.state,
+            form: {
+                label: '',
+                value: ''
+            },
+            store: dataStore
+        };
+    },
+    computed: {
+        activeSilences() {
+            return this.$root.activeSilences || [];
+        },
+        filteredSilences() {
+            if (!this.state.labels || this.state.labels.length === 0) {
+                return this.activeSilences;
+            }
+
+            return this.activeSilences.filter(silence => {
+                return this.state.labels.some(filterLabel => {
```

Comment:
```suggestion
                return this.state.labels.every(filterLabel => {
```

If we have multiple filters, we want all of them to match. This is the same behavior that Alertmanager has.

Reply:
Thank you. 
And for this one
>Regarding Git stuff, please put in separate commit all the changes related to the renaming of silence-modal into silence-create-modal.

Do I need to make a different PR for that?

Reply:
> Is it better to make a different PR for that?

No. Even if they are unrelated changes, it is totally fine to have a refactor commit in this PR, so your actual changes will fit better.

So, having one commit for the rename, and another commit for all the changes related to the new silence list modal is fine.

Reply:
No problem. Updated.

Reply:
I have tested it in my local environment and works well :+1: 

Nice job :)